pkg/monitor/intervalcreation/node_test.go

```diff
@@ -10,6 +11,191 @@ import (
+				},
+				From: time.UnixMicro(1664269199853216).In(time.UTC),
+				To:   time.UnixMicro(1664269199853216).In(time.UTC),
+			},
+		},
+		{
+			name:    "simple failure",
+			logLine: `Jul 05 17:47:52.807876 ci-op-lxqqvl5x-d3bee-gl4hp-master-0 hyperkube[1495]: I0606 17:47:52.807876    1599 prober.go:121] "Probe failed" probeType="Readiness" pod="openshift-authentication/oauth-openshift-77f7b95df5-r4xf7" podUID=1af660b3-ac3a-4182-86eb-2f74725d8415 containerName="oauth-openshift" probeResult=failure output="Get \"https://10.129.0.12:6443/healthz\": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"`,
+			want: monitorapi.EventInterval{
+				Condition: monitorapi.Condition{
+					Level:   0,
+					Locator: "ns/openshift-authentication pod/oauth-openshift-77f7b95df5-r4xf7 uid/1af660b3-ac3a-4182-86eb-2f74725d8415 container/oauth-openshift",
+					Message: "reason/ReadinessFailed Get \"https://10.129.0.12:6443/healthz\": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)",
+				},
+				From: time.UnixMicro(1657043272807876).In(time.UTC),
+				To:   time.UnixMicro(1657043272807876).In(time.UTC),
+			},
+		},
+		{
+			name:    "simple error",
+			logLine: `Jul 05 17:43:12.908344 ci-op-lxqqvl5x-d3bee-gl4hp-master-0 hyperkube[1495]: E0606 17:43:12.908344    1500 prober.go:118] "Probe errored" err="rpc error: code = NotFound desc = container is not created or running: checking if PID of 645437acbb2ca429c04d5a2628924e2e10d44c681c824dddc7c82ffa30a936be is running failed: container process not found" probeType="Readiness" pod="openshift-marketplace/redhat-operators-4jpg4" podUID=0bac4741-a3bd-483c-b119-e97663d64024 containerName="registry-server"`,
+			want: monitorapi.EventInterval{
+				Condition: monitorapi.Condition{
+					Level:   0,
+					Locator: "ns/openshift-marketplace pod/redhat-operators-4jpg4 uid/0bac4741-a3bd-483c-b119-e97663d64024 container/registry-server",
+					Message: "reason/ReadinessErrored rpc error: code = NotFound desc = container is not created or running: checking if PID of 645437acbb2ca429c04d5a2628924e2e10d44c681c824dddc7c82ffa30a936be is running failed: container process not found",
+				},
+				From: time.UnixMicro(1657042992908344).In(time.UTC),
+				To:   time.UnixMicro(1657042992908344).In(time.UTC),
+			},
+		},
+	}
+
+	logString := ""
+	for i := range testcase {
+
+		logString += testcase[i].logLine + "\n"
+	}
+
+	intervals := eventsFromKubeletLogs("testName", []byte(logString))
+
+	if intervals == nil {
+		t.Fatal("Invalid intervals")
+	}
+
+	if intervals.Len() != len(testcase) {
+		t.Fatal("Mismatched interval count")
+	}
+
+	for i := range intervals {
+		if !reflect.DeepEqual(intervals[i], testcase[i].want) {
```

Comment:
This is not consistently used here and the OGs of this repo probably think it's sacrilige, but I love the testity assert/require modules: https://github.com/stretchr/testify

They can make assertions more brief and give more consistent output, easier to view diffs, etc.

Not at all required for this PR, but can be very useful for testing in future.